### PR TITLE
BWS-179: + apply-pagination-hook-for-all-tables

### PR DESCRIPTION
**Pull request by ania-leichenko**
Screenshot_1
Screenshot_2
Screenshot_3
Screenshot_4
Screenshot_5

### CODE REVIEW

File: frontend/src/components/bs/components/spaces-table/spaces-table.tsx [hunk 1]
```diff
@@ -1,20 +1,52 @@
...
+import {
+  useAppSelector,
+  useMemo,
+  useAppDispatch,
+  usePagination,
+} from 'hooks/hooks';
 import { Table } from 'components/common/common';
 import { getRows, getColumns } from './helpers/helpers';
 import { DataStatus } from 'common/enums/enums';
+import { Pagination } from 'common/enums/enums';
```
Comment (Donymak) [inline frontend/src/components/bs/components/spaces-table/spaces-table.tsx]:
```suggestion
import { DataStatus, Pagination  } from 'common/enums/enums';
```
in every service table component

File: backend/src/data/repositories/slc-function/slc-function.repository.ts [hunk 2]
```diff
@@ -90,6 +90,20 @@ class SLCFunction {
     });
   }
 
+  public async getCount(filter: SLCFunctionGetFilter): Promise<number> {
+    const { tenantId } = filter;
+
+    return await this.#SLCFunctionModel
```
Comment (what1s1ove) [inline backend/src/data/repositories/slc-function/slc-function.repository.ts]:
```suggestion
  public getCount(filter: SLCFunctionGetFilter): Promise<number> {
    const { tenantId } = filter;

    return this.#SLCFunctionModel
```

File: backend/src/data/repositories/group/group.repository.ts [hunk 3]
```diff
@@ -73,6 +73,18 @@ class Group {
       .first();
   }
 
+  public async getCount(
```
Comment (what1s1ove) [inline backend/src/data/repositories/group/group.repository.ts]:
```suggestion
  public getCount(
```

File: backend/src/data/repositories/instance/instance.repository.ts [hunk 4]
```diff
@@ -65,6 +65,14 @@ class Instance {
       .limit(limit);
   }
 
+  public async getCount(tenantId: string): Promise<number> {
```
Comment (what1s1ove) [inline backend/src/data/repositories/instance/instance.repository.ts]:
```suggestion
  public getCount(tenantId: string): Promise<number> {
```

File: frontend/src/components/bs/components/spaces-table/spaces-table.tsx (hunk 1 above)
Comment (what1s1ove) [reply frontend/src/components/bs/components/spaces-table/spaces-table.tsx]: agreed

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**